src/Microsoft.Identity.Client/TokenCache.cs

```diff
@@ -260,7 +260,9 @@ internal RefreshTokenCacheItem FindRefreshToken(AuthenticationRequestParameters
         {
             lock (LockObject)
             {
-                TokenCacheKey key = new TokenCacheKey(null, null, requestParam.ClientId, requestParam.User?.HomeObjectId);
+                RefreshTokenCacheKey key = new RefreshTokenCacheKey(
+                    new Uri(requestParam.Authority.CanonicalAuthority).Host, requestParam.ClientId,
```

Comment:
fixed

---
In reply to: [109345765](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/282#discussion_r109345765) [](ancestors = 109345765)